dashboard/app/models/user.rb

```diff
@@ -157,6 +157,12 @@ class User < ActiveRecord::Base
   accepts_nested_attributes_for :school_info, reject_if: :preprocess_school_info
   validates_presence_of :school_info, unless: :school_info_optional?
 
+  validate :must_be_a_teacher_if_has_sections, on: :update, if: -> {user_type_changed? && student?}
```

Comment:
Has this been cleared by product? A "typical" case may be a student making themselves a teacher, screwing around (creating sections), then trying to return to be a student.